webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js

```diff
@@ -14,7 +14,11 @@ const SettingCellInner = props => {
   });
 
   const field = (
-    <span {...rest} className={cssClasses} onClick={() => onEditClick(setting)}>
+    <span
+      {...rest}
+      className={cssClasses}
+      onClick={() => !setting.readonly && onEditClick(setting)}
```

Comment:
This adds a redundant onClick in read-only fields.
How about using a conditional prop rendering? 

```js
  const onClickHandler = {
    ...(!setting.readonly && { onClick: () => onEditClick(setting) }),
  };

  const field = (
    <span {...rest} className={cssClasses} {...onClickHandler}>
      {valueToString(setting)}
    </span>
  );
```

Reply:
Thanks for the suggestion! I actually took it one step further in the second commit and moved the entire edit handling into the cell, lmk what you think about it (i'm guessing tests/lint will fail, but didn't want to invest in fixing them before getting feedback on the general approach)